controllers/fenceagentsremediation_controller.go

```diff
@@ -132,13 +132,13 @@ func (r *FenceAgentsRemediationReconciler) Reconcile(ctx context.Context, req ct
 		return emptyResult, err
 	}
 
-	// Check NHC timeout annotation
-	if isTimedOutByNHC(far) {
+	// Check NHC timeout annotation and stop the agent, since remediation is no longer relevant (most likely because fixed by a different remediator)
+	if r.Executor.Exists(far.GetUID()) && isTimedOutByNHC(far) {
```

Comment:
you add a check here, and few lines below you requeue.... I think that's a significant unwanted change in execution flow, what happens on next reconcile?